tutorials/Project.toml

```diff
@@ -1,8 +1,16 @@
 [deps]
+BoundaryValueDiffEq = "764a87c0-6b3e-53db-9096-fe964310641d"
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+DiffEqCallbacks = "459566f4-90b8-5000-8ac3-15dfb0a30def"
 IJulia = "7073ff75-c697-5162-941a-fcdaad2a7d2a"
 Manifolds = "1cead3c2-87b3-11e9-0ccd-23c62b72b94e"
 ManifoldsBase = "3362f125-f0bb-47a3-aa74-596ffd7ef2fb"
 Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
+MultivariateStats = "6f286f6a-111f-5878-ab1e-185364afe411"
+OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+RecursiveArrayTools = "731186ca-8d62-57ce-b412-fbd966d074cd"
 
 [compat]
```

Comment:
Sure, I think the relevant dependencies are `CSV`, `DataFrames` and `MultivariateStats`.

Reply:
If you could do it for all, that would make sure that the tutorials basically never break even with updates of the packages.

Reply:
I think other dependencies are constrained transitively by Manifolds.jl anyway?

Reply:
Hm, I am not sure about that.

Reply:
I'd like to avoid putting too much here because I'm afraid we will just forget to update this file, and CompatHelper doesn't seem to work for such Project.toml files: https://github.com/JuliaRegistries/CompatHelper.jl/issues/408 .

Reply:
We might have to think about those ourselves, yes. I 'd like to make at least sure that the tutorials – they are cached and will not run that often – suddenly fail to run, because we forgot a compat.
Also with the cache we use for the tutorials CompatHelper would not note that as well.

Reply:
Tutorials can also fail to run because we they have obsolete compat entries. Unless we want to render tutorials with older versions of Manifolds.jl/ManifoldsBase.jl than the newest one? That doesn't seem ideal either. I think ideally we should have a loud and as early as possible error when tutorials stop working with the latest Manifolds.jl. I would prefer to avoid merging PRs that break tutorials.

Reply:
That is manually done by having Manifolds in dev mode from the parent folder when setting everything up in the make.jl, so that we run on the most recent one, yes.
Hm, now I am unsure, because what I want to avoid is that someone wants to contribute and their PR fails because some tutorial fails (maybe they fixed a typo in there). But maybe there is no real solution to this.

Reply:
> Hm, now I am unsure, because what I want to avoid is that someone wants to contribute and their PR fails because some tutorial fails (maybe they fixed a typo in there). But maybe there is no real solution to this.

The only solution to this problem is specifying exact versions of dependencies in Project.toml. Then nothing can break when someone contributes a typo fix. When we allow "non-breaking" updates of dependencies, CI can randomly break. And sometimes it does. I don't think that has ever been a big issue? We either add an upper bound to the dependency ourselves or wait a bit until it's fixed.

Reply:
Well, ok, then let's leave it as is.